### ISSUE TITLE
Seasonality of weekly data: what period should be applied

### ISSUE THREAD

**Issue by slavakx** [facebook/prophet#2112]
Hi, 

If I have weekly data, should I apply a custom seasonality like:
 
`
add_seasonality(m, name='yearly', period=365.25/7, fourier.order=5)
`

or use default parameter 
`
Prophet(yearly_seasonality=True)
`

The reason I ask is that the default parameter has always a yearly period of 365.25 days

Thanks

**Comment by tcuongd** [comment]
The underlying model is continuous-time, so it should handle your data without needing adjustments. Just make sure that your prediction dataframe only contains one day a week (and is indexed to the same day as your training data). See https://facebook.github.io/prophet/docs/non-daily_data.html#monthly-data

**Comment by Serge9744** [comment]
Hi ,
I would have one question though , when you say the model is continuous , how can it be?
In the article you use a matrix X which is based on the data points at time t or maybe i do not understand that.
Do you interpolate the points to a higher frequency ?
For example if I have a weekly seasonality normally i should not use a frequency which is lower or equal to twice a month as the data is sampled weekly.
But under the hood do you interpolate the time series to daily points?
In an experiment with weekly data I have force the weekly seasonality to 8 coefficients and it increases drastically the performances on an unseen validation set ( from 0.3 R2 to 0.6 without weekly and with weekly seasonality which for me doesn't make sense).
I of course predict on weekly data too
Is it possible to help me understand please?
